src/dataset.rs

```diff
@@ -544,7 +544,10 @@ impl Dataset {
     ///
     /// Applies to raster datasets, and fetches the
     /// rasterband at the given _1-based_ index.
-    pub fn rasterband(&self, band_index: isize) -> Result<RasterBand> {
+    pub fn rasterband(&self, band_index: usize) -> Result<RasterBand> {
+        assert!(band_index > 0);
```

Comment:
i think we should return an error here.

Reply:
The method already returns a Result so why panic when we can also return a matching error?

Reply:
Because it's not an error the caller can handle, it's a logic bug in the code. So it makes sense to report it right away instead of waiting until later, when it might be less clear what failed.

Reply:
thats a good idea. thanks!

Reply:
It reports an error like `ERROR 5: foo.tif: GDALDataset::GetRasterBand(2) - Illegal band #`.

For 0 we could just call `GDALGetRasterBand`, but for a band number larger than `c_int::MAX` we'd need to synthetize the message or add a new error variant.

Reply:
if it reports an error for 0 then its fine for me

Reply:
Updated.